setup.py

```diff
@@ -1,31 +1,99 @@
+]
 
-with open("README.md") as f:
-    readme = f.read()
-
-setuptools.setup(
-    name="jupyter-server-proxy",
-    version='1.6.0',
-    url="https://github.com/jupyterhub/jupyter-server-proxy",
-    author="Ryan Lovett & Yuvi Panda",
-    author_email="rylo@berkeley.edu",
-    license="BSD 3-Clause",
-    description="Jupyter server extension to supervise and proxy web services",
-    long_description=readme,
+cmdclass = create_cmdclass(
+    "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE / "jupyterlab-server-proxy", build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "jupyterlab-server-proxy" / "package.json").read_bytes())
+
+setup_args = dict(
+    name=name,
+    version=pkg_json["version"],
+    url=pkg_json["homepage"],
+    author=pkg_json["author"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
+    long_description=long_description,
     long_description_content_type="text/markdown",
+    cmdclass=cmdclass,
     packages=setuptools.find_packages(),
-    install_requires=['notebook', 'simpervisor>=0.4', 'aiohttp'],
-    python_requires='>=3.5',
+    install_requires=["jupyterlab~=3.0", "notebook", "simpervisor>=0.4", "aiohttp"],
+    zip_safe=False,
+    include_package_data=True,
+    python_requires=">=3.6",
+    platforms="Linux, Mac OS X, Windows",
+    keywords=["Jupyter", "JupyterLab", "JupyterLab3"],
     classifiers=[
-        'Framework :: Jupyter',
+        "Framework :: Jupyter",
     ],
     data_files=[
-        ('share/jupyter/nbextensions/jupyter_server_proxy', glob('jupyter_server_proxy/static/*')),
-        ('etc/jupyter/jupyter_notebook_config.d', ['jupyter_server_proxy/etc/jupyter-server-proxy-notebookserverextension.json']),
-        ('etc/jupyter/jupyter_server_config.d', ['jupyter_server_proxy/etc/jupyter-server-proxy-jupyterserverextension.json']),
-        ('etc/jupyter/nbconfig/tree.d', ['jupyter_server_proxy/etc/jupyter-server-proxy-nbextension.json'])
+        ("share/jupyter/nbextensions/jupyter_server_proxy", glob("jupyter_server_proxy/static/*")),
+        (
+            "etc/jupyter/jupyter_notebook_config.d",
+            ["jupyter_server_proxy/etc/jupyter-server-proxy-notebookserverextension.json"],
+        ),
+        (
+            "etc/jupyter/jupyter_server_config.d",
+            ["jupyter_server_proxy/etc/jupyter-server-proxy-jupyterserverextension.json"],
+        ),
+        (
+            "etc/jupyter/nbconfig/tree.d",
+            ["jupyter_server_proxy/etc/jupyter-server-proxy-nbextension.json"],
+        ),
```

Comment:
I believe that the `data_files` property here is redundant with the `data_files_spec` variable. However, I'm not super sure what it does, so I decided to keep both. If anybody knows if and how you can merge those two, please let me know.

Reply:
One way would be to move these to `data_files_spec` so it's all handled by `jupyter-packaging`. Although the format would be slightly different.

As an example: https://github.com/jupyter-server/jupyter-resource-usage/blob/ef909637344d97b84a650a0dbe4f0487fb403bb7/setup.py#L29-L48

Reply:
Thanks for the hint, @jtpio! I adjusted it in https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/280ce015b05732438f42998318960752085f6b25. Could you maybe have another look, please?